packages/nimble-components/src/rich-text/models/tests/markdown-parser.spec.ts

```diff
@@ -923,4 +928,317 @@ describe('Markdown parser', () => {
+                        '^user:.*'
+                    ));
+                    await connect();
+                    mentionsMap.set(
+                        element.mentionInternals.character,
+                        element.mentionInternals
+                    );
+                    const doc = RichTextMarkdownParser.parseMarkdownToDOM(
+                        '<user:1>',
+                        mentionsMap
+                    );
+
+                    expect(getTagsFromElement(doc)).toEqual([
+                        'P',
+                        `${richTextMentionUsersViewTag}`.toUpperCase()
+                    ]);
+                    expect(
+                        getLastChildElementAttribute('mention-label', doc)
+                    ).toEqual(name);
+                });
+            });
+        });
+
+        it('should get anchor element when markdown input does not match with the pattern', async () => {
+            ({ element, connect, disconnect } = await setup(
+                [
+                    { key: 'user:1', displayName: 'username1' },
+                    { key: 'user:2', displayName: 'username2' }
+                ],
+                '^user:.*'
+            ));
+            await connect();
+            mentionsMap.set(
+                element.mentionInternals.character,
+                element.mentionInternals
+            );
+            const doc = RichTextMarkdownParser.parseMarkdownToDOM(
+                '<https://1>',
+                mentionsMap
+            );
+
+            expect(getTagsFromElement(doc)).toEqual([
+                'P',
+                `${anchorTag}`.toUpperCase()
+            ]);
+            expect(getLastChildElementAttribute('href', doc)).toEqual(
+                'https://1'
+            );
+        });
+
+        it('should get anchor element when display name is missing', async () => {
+            ({ element, connect, disconnect } = await setup(
+                [{ key: 'user:1' }, { key: 'user:2' }],
+                '^user:.*'
+            ));
+            await connect();
+            mentionsMap.set(
+                element.mentionInternals.character,
+                element.mentionInternals
+            );
+            const doc = RichTextMarkdownParser.parseMarkdownToDOM(
+                '<user:1>',
+                mentionsMap
+            );
+
+            expect(getTagsFromElement(doc)).toEqual([
+                'P',
+                `${anchorTag}`.toUpperCase()
+            ]);
+            expect(lastChildElementHasAttribute('href', doc)).toBeFalse();
+        });
+
+        it('should get anchor element when pattern is not a valid Regex', async () => {
+            ({ element, connect, disconnect } = await setup(
+                [
+                    { key: 'user:1', displayName: 'username1' },
+                    { key: 'user:2', displayName: 'username1' }
+                ],
+                'abc'
```

Comment:
'abc' is a valid regex. Really this test case is covering the case where the absolute link is not http/https, and it doesn't match the regex. Maybe just change the test name.

Reply:
Updated the test name